code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,575 @@
+            } else {
+                MobileCore.log(
+                    LoggingMode.VERBOSE,
+                    TAG,
+                    "Failed to download configuration. Applying Will retry download."
+                )
+                // If the configuration download fails, publish current configuration and retry download again.
+                publishConfigurationState(
+                    configurationStateManager.environmentAwareConfiguration,
+                    event
+                )
+                retryConfigTaskHandle = retryConfigDownload(appId)
+            }
+
+            // Start event processing again
+            api.startEvents()
+        }
+    }
+
+    /**
+     * Retrieves the configuration from the file path specified in the [event]'s data and replaces
+     * the current configuration with it.
+     *
+     * @param event the trigger event (whose event data contains the file path for retrieving the config)
+     *              requesting a configuration change
+     */
+    private fun configureWithFilePath(event: Event) {
+        val filePath =
+            event.eventData?.get(Configuration.CONFIGURATION_REQUEST_CONTENT_JSON_FILE_PATH) as String?
+
+        if (filePath.isNullOrBlank()) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Unable to read config from provided file (filePath is invalid)"
+            )
+            return
+        }
+
+        val config = configurationStateManager.getConfigFromFile(filePath)
+        if (config == null) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Unable to read config from provided file (content is invalid)"
+            )
+            return
+        }
+
+        configurationStateManager.replaceConfiguration(config)
+        applyConfigurationChanges(event, RulesSource.REMOTE)
+    }
+
+    /**
+     * Updates the current configuration with the content from a file asset.
+     *
+     * @param event which contains [Configuration.CONFIGURATION_REQUEST_CONTENT_JSON_ASSET_FILE] as part of its event data
+     */
+    private fun configureWithFileAsset(event: Event) {
+        val fileAssetName =
+            event.eventData?.get(Configuration.CONFIGURATION_REQUEST_CONTENT_JSON_ASSET_FILE) as String?
+
+        if (fileAssetName.isNullOrBlank()) {
+            MobileCore.log(
+                LoggingMode.DEBUG,
+                TAG,
+                "Asset file name for configuration is null or empty."
+            )
+            return
+        }
+
+        val config = configurationStateManager.loadBundledConfig(fileAssetName)
+
+        if (config.isNullOrEmpty()) {
+            MobileCore.log(
+                LoggingMode.DEBUG,
+                TAG,
+                "Empty configuration found when processing JSON string."
+            )
+            return
+        }
+
+        configurationStateManager.replaceConfiguration(config)
```

Comment:
Can you design this api similar to `updateConfigWithAppId` and keep replaceConfigration internal to ConfigurationStateManager